### ISSUE TITLE
Type Errors do not give helpfull error messages

### ISSUE THREAD

**Issue by IsaacOscar** [gracelang/minigrace#280]
Error messages given for type errors are not very usefull, e.g:
```
TypeError: result of method foo(_) does not have type Bar.
It is missing method bar.
```
This dosn't tell me what it is that ~~was passed to~~ the foo(_) method _is trying to return_.

I have "fixed" this by adding the indicated line to "/usr/lib/grace/modules/gracelib.js":

```javascript
function raiseTypeError(msg, type, value) {
    var diff;
    if (GraceDone === value) {
        diff = " — it is `done`.";
    } else {
        var mm = do_import("mirrors", gracecode_mirrors);
        try {
             var tc = callmethod(mm, "loadDynamicModule(1)", [1], new GraceString("typeComparison"));
             var missing = callmethod(tc, "methodsIn(1)missingFrom(1)", [1, 1], type, value)._value;
             var s = (missing.includes(" ")) ? "s " : " ";
             diff = ".\nIt is missing method" + s + missing + "." +
/*ADDED--------->*/"\nThe object was created by \"" + value.className + "\" in module \"" + value.definitionModule + "\" (line " + value.definitionLine + ").";
        } catch (ex) {
             // if something goes wrong while generating the message, just give up
        }
        if (! diff) { diff = ""; }
    }
    var ex = new GraceExceptionPacket(TypeErrorObject,
                                      new GraceString(msg + diff));
    throw ex;
}
```
I havn't added it to the source code as there are 9 files definining that function, and I don't know which is the appropriate one to modify.

Now it will also print
```
The object was created by "baz(_)" in module "Baz" (line 3).
```

Indicating "what" the object is (i.e. a Baz).
This is very helpfull for debugging grace programs!

**Comment by apblack** [comment]
This is a good suggestion.   There is already a JavaScript function in _gracelib.js_ called `describe` that attempts to build a useful description of an arbitrary Grace object.  Do you think that it should add the definition module and line number?  This function could then be called from `raiseTypeError`.

The master version of  _gracelib.js_  is in the _js_ directory.  The other copies are made as part of the build process.

**Comment by apblack** [comment]
Solved by 79adeec9b8eee0a9472c9b6329202408916f6ec9
